### PR TITLE
Added two more infoblox DHCP options

### PR DESCRIPTION
**Pull request by lzap**
I am shooting in the dark here. Tests won't work locally.

### CODE REVIEW

**Comment by ekohl** [comment]
Needs https://github.com/theforeman/smart_proxy_dhcp_infoblox/pull/46

**Comment by lzap** [comment]
The other PR merged, rebased to trigger tests.

File: spec/classes/foreman_proxy__plugin__dhcp__infoblox_spec.rb [hunk 1]
```diff
@@ -40,6 +40,8 @@
             :record_type  => 'host',
             :dns_view     => 'non-default',
             :network_view => 'another-non-default',
+            ':wait_after_restart: 99',
+            ":options: '[]'",          ])
```
Comment (ekohl) [inline spec/classes/foreman_proxy__plugin__dhcp__infoblox_spec.rb:44]: Syntax error here. Probably the same on line 57

Comment (lzap) [reply spec/classes/foreman_proxy__plugin__dhcp__infoblox_spec.rb]: Sorry, fixed. It's just all red locally and I don't know why.

**Comment by mmoll** [comment]
Tests fail :broken_heart:

File: manifests/plugin/dhcp/infoblox.pp [hunk 2]
```diff
@@ -14,12 +14,18 @@
...
+# $options:: DHCP options in JSON format: [{"name": "","num": 1,"value": "","vendor_class": ""}]
+#
 class foreman_proxy::plugin::dhcp::infoblox (
   String $username = undef,
   String $password = undef,
   Enum['host', 'fixedaddress'] $record_type = 'fixedaddress',
   String $dns_view = 'default',
   String $network_view = 'default',
+  Integer $wait_after_restart = 10,
+  String $options = undef,
```
Comment (lzap) [inline manifests/plugin/dhcp/infoblox.pp:28]: I am going to make this `'[]'`.

Comment (lzap) [reply manifests/plugin/dhcp/infoblox.pp:28]: Can you help me how to refactor this if I default to "[]" here:

```
<% unless [nil, :undefined, :undef].include?(scope.lookupvar('::foreman_proxy::plugin::dhcp::infoblox::options')) -%>
```

Comment (mmoll) [reply manifests/plugin/dhcp/infoblox.pp:28]: Not 100% sure how this should be at the end... is the input a single string or an array of strings?

Comment (mmoll) [reply manifests/plugin/dhcp/infoblox.pp:28]: Id the default is just '[]' as string any thing can be set, just render out the variable.

Comment (ekohl) [reply manifests/plugin/dhcp/infoblox.pp:28]: Why do you use the String data type and not at least `Array[String]`? I'm not sure how Kafo would deal with `Array[Hash]`, but `Array[String]` should be fine.
```suggestion
  Array[String] $options = [],
```

Comment (lzap) [reply manifests/plugin/dhcp/infoblox.pp:28]: It is actually string with JSON, I will make it '[]' as the default.

File: templates/plugin/dhcp_infoblox.yml.erb [hunk 3]
```diff
@@ -6,10 +6,21 @@
...
+
+# View used for host record type, usually 'default.myview' for custom names.
+:dns_view: "<%= scope.lookupvar('::foreman_proxy::plugin::dhcp::infoblox::dns_view') %>"
+
+# Number of seconds to wait after successful Grid restart (make sure to increase Foreman and CLI timeouts).
+:wait_after_restart: <%= scope.lookupvar('::foreman_proxy::plugin::dhcp::infoblox::wait_after_restart') %>
+
+# JSON-encoded array of dicts with name, num, value and vendor_class keys for DHCP option(s).
+# Set to blank for no options, this is only used for 'fixedaddress' record types.
+:options: '<%= scope.lookupvar('::foreman_proxy::plugin::dhcp::infoblox::options') %>'
```
Comment (ekohl) [inline templates/plugin/dhcp_infoblox.yml.erb:26]: The file is also YAML. Why doesn't the plugin simply read structured data?

**Comment by lzap** [comment]
The reason for that is that we want users to be able to pass this via CLI. How you can pass YAML on a single line? And if you can do it, don't show it to me I don't want to even see this. JSON is much better IMHO.

> To reduce the possible issues, should we call parsejson to ensure it is valid?

If you ask me, I say no as I don't know how to do this. I can make an attempt to do this: https://stackoverflow.com/questions/43729915/parse-a-json-string-in-puppet but unsure how to handle exceptions etc.

**Comment by lzap** [comment]
Can we move forward with this one? To me it looks like YAML with JSON mix is a good solution to the problem.

**Comment by lzap** [comment]
Closing, feel free to take it from here if you think this patch is useful. Cheers!

**Pull request closed without merging**